src/test/java/io/stargate/sgv2/jsonapi/testresource/DseTestResource.java

```diff
@@ -89,6 +90,14 @@ public Map<String, String> start() {
     Map<String, String> env = super.start();
     ImmutableMap.Builder<String, String> propsBuilder = ImmutableMap.builder();
     propsBuilder.putAll(env);
+
+    // 02-April-2025, yuqi: [data-api#1972] Set the system property variable to indicate
+    // integration test.
+    // This is helpful with testing customized provider configuration. E.G. model deprecation. See
+    // detail in
+    // EmbeddingConfigSourceProvider.java
+    propsBuilder.put(EmbeddingConfigSourceProvider.DATA_API_INTEGRATION_TEST, "true");
```

Comment:
We should probably make this more general, to be usable by other integration tests where we must... but not sure where to define it (because it needs to be accessible from non-test code too)